### PR TITLE
Fix all the nodeSelectors and tolerations

### PR DESCRIPTION
**Pull request by aiyengar2**
Found errors on nodeSelectors and tolerations throughout all the charts, so I standardized them to all do the same thing (use the helper template and always add Windows tolerations and nodeSelectors). The only charts I did not touch are longhorn (not owned by our org), harvester (not owned by our org), and rancher-webook (@MbolotSuse is making the webhook PR).

Also removed the Chart.yaml.orig from the rancher-backup chart and moved gitjob into it's own package.

Will go back and contribute PRs for the changes to the relevant repositories.

Tested using variations of `helm template test packages/${PACKAGE}/charts --set nodeSelector.hi=bye --set nodeSelector.bye=hi --set 'tolerations={hi,bye}'` and a `helm template` without any sets outside the default in order to confirm that nodeSelectors and tolerations were applied. Also inspected all the templates and found some additional stuff that needed to be added.

### CODE REVIEW

File: packages/fleet/gitjob/package.yaml
Comment (aiyengar2) [inline packages/fleet/gitjob/package.yaml:3]: cc: @prachidamle

File: packages/fleet/gitjob/generated-changes/patch/values.yaml.patch
Comment (aiyengar2) [inline packages/fleet/gitjob/generated-changes/patch/values.yaml.patch:14]: this was modified to keep it consistent with the others

File: packages/fleet/fleet/generated-changes/patch/values.yaml.patch
Comment (aiyengar2) [inline packages/fleet/fleet/generated-changes/patch/values.yaml.patch:14]: this was modified to keep it consistent with the others

File: packages/fleet/fleet-agent/generated-changes/patch/values.yaml.patch
Comment (aiyengar2) [inline packages/fleet/fleet-agent/generated-changes/patch/values.yaml.patch:33]: this was modified to keep it consistent with the others

File: packages/rancher-aks-operator/rancher-aks-operator/generated-changes/patch/templates/deployment.yaml.patch
Comment (aiyengar2) [inline packages/rancher-aks-operator/rancher-aks-operator/generated-changes/patch/templates/deployment.yaml.patch:20]: was broken when you tried to supply nodeSelectors or tolerations

File: packages/rancher-alerting/rancher-sachet/charts/values.yaml
Comment (aiyengar2) [inline packages/rancher-alerting/rancher-sachet/charts/values.yaml:65]: already has support for using the helper template to auto-insert these tolerations and nodeSelectors, not sure why it was added twice

File: packages/rancher-alerting/rancher-prom2teams/generated-changes/patch/values.yaml.patch
Comment (aiyengar2) [inline packages/rancher-alerting/rancher-prom2teams/generated-changes/patch/values.yaml.patch:47]: already has support for using the helper template to auto-insert these tolerations and nodeSelectors, not sure why it was added twice

File: packages/rancher-alerting/rancher-alerting-drivers/charts/values.yaml
Comment (aiyengar2) [inline packages/rancher-alerting/rancher-alerting-drivers/charts/values.yaml:27]: added the ability to specify nodeSelectors and tolerations for good measure.

File: packages/rancher-backup/rancher-backup/generated-changes/overlay/Chart.yaml.orig
Comment (aiyengar2) [inline packages/rancher-backup/rancher-backup/generated-changes/overlay/Chart.yaml.orig:7]: @superseb done in this PR

File: packages/rancher-backup/rancher-backup/generated-changes/patch/templates/hardened.yaml.patch
Comment (aiyengar2) [inline packages/rancher-backup/rancher-backup/generated-changes/patch/templates/hardened.yaml.patch:13]: nodeSelectors and tolerations were missing

File: packages/rancher-backup/rancher-backup/generated-changes/patch/templates/deployment.yaml.patch
Comment (aiyengar2) [inline packages/rancher-backup/rancher-backup/generated-changes/patch/templates/deployment.yaml.patch:28]: was broken when you tried to supply nodeSelectors or tolerations

File: packages/rancher-cis-benchmark/charts/values.yaml
Comment (aiyengar2) [inline packages/rancher-cis-benchmark/charts/values.yaml:34]: this was modified to keep it consistent with the others

File: packages/rancher-eks-operator/rancher-eks-operator/generated-changes/patch/templates/deployment.yaml.patch
Comment (aiyengar2) [inline packages/rancher-eks-operator/rancher-eks-operator/generated-changes/patch/templates/deployment.yaml.patch:20]: was broken when you tried to supply nodeSelectors or tolerations

File: packages/rancher-gatekeeper/generated-changes/patch/templates/gatekeeper-audit-deployment.yaml.patch
Comment (aiyengar2) [inline packages/rancher-gatekeeper/generated-changes/patch/templates/gatekeeper-audit-deployment.yaml.patch:22]: if you supplied the nodeSelector in the original solution, it would not contain the selectors for Windows.

File: packages/rancher-gatekeeper/generated-changes/patch/templates/gatekeeper-audit-deployment.yaml.patch
Comment (aiyengar2) [inline packages/rancher-gatekeeper/generated-changes/patch/templates/gatekeeper-audit-deployment.yaml.patch:33]: tolerations were missing

File: packages/rancher-gatekeeper/generated-changes/patch/templates/gatekeeper-controller-manager-deployment.yaml.patch
Comment (aiyengar2) [inline packages/rancher-gatekeeper/generated-changes/patch/templates/gatekeeper-controller-manager-deployment.yaml.patch:22]: if you supplied the nodeSelector in the original solution, it would not contain the selectors for Windows.

File: packages/rancher-gatekeeper/generated-changes/patch/templates/gatekeeper-controller-manager-deployment.yaml.patch
Comment (aiyengar2) [inline packages/rancher-gatekeeper/generated-changes/patch/templates/gatekeeper-controller-manager-deployment.yaml.patch:33]: tolerations were missing

File: packages/rancher-gatekeeper/generated-changes/patch/templates/namespace-post-install.yaml.patch
Comment (aiyengar2) [inline packages/rancher-gatekeeper/generated-changes/patch/templates/namespace-post-install.yaml.patch:10]: tolerations were missing, nodeSelector was hard-coded

File: packages/rancher-gatekeeper/generated-changes/patch/templates/upgrade-crds-hook.yaml.patch
Comment (aiyengar2) [inline packages/rancher-gatekeeper/generated-changes/patch/templates/upgrade-crds-hook.yaml.patch:18]: tolerations were missing, nodeSelector was hard-coded

File: packages/rancher-gatekeeper/generated-changes/patch/values.yaml.patch
Comment (aiyengar2) [inline packages/rancher-gatekeeper/generated-changes/patch/values.yaml.patch:48]: this was modified to keep it consistent with the others

File: packages/rancher-gatekeeper/templates/crd-template/values.yaml
Comment (aiyengar2) [inline packages/rancher-gatekeeper/templates/crd-template/values.yaml]: this was modified to keep it consistent with the others

File: packages/rancher-gke-operator/rancher-gke-operator/generated-changes/patch/templates/deployment.yaml.patch
Comment (aiyengar2) [inline packages/rancher-gke-operator/rancher-gke-operator/generated-changes/patch/templates/deployment.yaml.patch:20]: was broken when you tried to supply nodeSelectors or tolerations

File: packages/rancher-istio/rancher-tracing/charts/templates/deployment.yaml
Comment (aiyengar2) [inline packages/rancher-istio/rancher-tracing/charts/templates/deployment.yaml:84]: nodeSelectors were missing, tolerations were broken.

note: affinity here only does arch, not OS

File: packages/rancher-istio/rancher-tracing/charts/values.yaml
Comment (aiyengar2) [inline packages/rancher-istio/rancher-tracing/charts/values.yaml:8]: this was modified to keep it consistent with the others

File: packages/rancher-istio/rancher-kiali-server/generated-changes/patch/templates/deployment.yaml.patch
Comment (aiyengar2) [inline packages/rancher-istio/rancher-kiali-server/generated-changes/patch/templates/deployment.yaml.patch:58]: was broken when you tried to supply nodeSelectors or tolerations

File: release.yaml
Comment (aiyengar2) [inline release.yaml:35]: need to bump since the earlier version was already released

File: packages/rancher-istio/rancher-istio/charts/configs/istio-base.yaml
Comment (aiyengar2) [inline packages/rancher-istio/rancher-istio/charts/configs/istio-base.yaml:7]: all of these components require nodeSelectors and tolerations, which can be supplied if you look at the IstioOperator spec: https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#IstioOperatorSpec

Currently, there were no nodeSelectors or tolerations

File: packages/rancher-istio/rancher-istio/charts/values.yaml
Comment (aiyengar2) [inline packages/rancher-istio/rancher-istio/charts/values.yaml:98]: this was modified to keep it consistent with the others

File: packages/rancher-sriov/generated-changes/patch/values.yaml.patch
Comment (aiyengar2) [inline packages/rancher-sriov/generated-changes/patch/values.yaml.patch:18]: this was modified to keep it consistent with the others

File: packages/rancher-vsphere/rancher-vsphere-cpi/generated-changes/patch/templates/daemonset.yaml.patch
Comment (aiyengar2) [inline packages/rancher-vsphere/rancher-vsphere-cpi/generated-changes/patch/templates/daemonset.yaml.patch:8]: if nodeSelectors aren't supplied, the affinity rules were fine.

However, if nodeSelectors are supplied, we should still work on a Windows cluster.

File: packages/rancher-vsphere/rancher-vsphere-cpi/generated-changes/patch/templates/daemonset.yaml.patch
Comment (aiyengar2) [inline packages/rancher-vsphere/rancher-vsphere-cpi/generated-changes/patch/templates/daemonset.yaml.patch:17]: missing tolerations

File: packages/rancher-vsphere/rancher-vsphere-cpi/generated-changes/patch/values.yaml.patch
Comment (aiyengar2) [inline packages/rancher-vsphere/rancher-vsphere-cpi/generated-changes/patch/values.yaml.patch:8]: seems like a typo

File: packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/templates/controller/deployment.yaml.patch
Comment (aiyengar2) [inline packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/templates/controller/deployment.yaml.patch:8]: if nodeSelectors are not supplied, this code worked by using affinity.

If nodeSelectors are supplied, this code was broken on Windows clusters since default node-selectors are not applied for Windows.

File: packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/templates/controller/deployment.yaml.patch
Comment (aiyengar2) [inline packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/templates/controller/deployment.yaml.patch:17]: missing tolerations

File: packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/templates/node/daemonset.yaml.patch
Comment (aiyengar2) [inline packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/templates/node/daemonset.yaml.patch:8]: if nodeSelectors are not supplied, this code worked by using affinity.

If nodeSelectors are supplied, this code was broken on Windows clusters since default node-selectors are not applied for Windows.

File: packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/templates/node/daemonset.yaml.patch
Comment (aiyengar2) [inline packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/templates/node/daemonset.yaml.patch:17]: missing tolerations

File: packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/templates/node/daemonset.yaml.patch
Comment (aiyengar2) [inline packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/templates/node/daemonset.yaml.patch:20]: nit: indentation was inconsistent, with 6 it looks like:
```
tolerations:
- hi
```

v.s.

```
tolerations:
  - hi
```

Since I'm using 8 around other charts and we use 8 in the csiController anyways, I standardized to 8.

File: packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/values.yaml.patch
Comment (aiyengar2) [inline packages/rancher-vsphere/rancher-vsphere-csi/generated-changes/patch/values.yaml.patch:8]: nit: leave it empty and the outcome is the same.

**Comment by aiyengar2** [comment]
Now that all the changes are tracked here, I will reverse submit PRs to the upstreams in order to absorb the changes introduced in this PR.

However, I will **not** be changing the contents of this PR as much as possible. Instead, when I modify the upstream, I'll just add a commit to the end that modifies **just packages/**; if the chart produced is the same, there should be no need to run `make charts`.

**Comment by aiyengar2** [comment]
Added @manuelbuil to review sriov changes only

Added @PennyScissors to review Istio (I know the chart enough, like Monitoring and Alerting, but just want a second opinion in case there's something I missed cause I'm not super familiar)

Added @prachidamle for CIS Benchmark

**Comment by aiyengar2** [comment]
> Now that all the changes are tracked here, I will reverse submit PRs to the upstreams in order to absorb the changes introduced in this PR.
> 
> ~However, I will **not** be changing the contents of this PR as much as possible. Instead, when I modify the upstream, I'll just add a commit to the end that modifies **just packages/**; if the chart produced is the same, there should be no need to run `make charts`.~
> 
> Edit: This is not possible since we need to bump RC versions. I will need to force push the changes.

I'll clean up the PR once all the upstream changes are in.

**Comment by aiyengar2** [comment]
After talking to @luthermonson and @rosskirkpat offline, I will not be rebasing this PR on the changes introduced into upstream in https://github.com/rancher/vsphere-charts/pull/17 since there are other changes that need to go into upstream before we modify what is in rancher/charts accordingly.

The Windows team will file a PR later to rebase against the new upstream once all PRs are merged.

**Review by PennyScissors — APPROVED** [review]

**Review by prachidamle — APPROVED** [review]

**Pull request merged**
